### PR TITLE
Feature/tooltip rework

### PR DESCRIPTION
**Pull request by NiclasNorin**

### CODE REVIEW

File: source/js/tooltip.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    handleClasses(element, removeClasses, addClass = false) {
+        removeClasses.forEach(className => {
+            element.classList.remove(className);
+        });
+
+        addClass && element.classList.add(addClass);
+
+    }
+}
+export default Tooltip; */
```
Comment (sebastianthulin) [inline source/js/tooltip.js]: Remove commented code.

File: source/sass/component/_tooltip.scss [hunk 2]
```diff
@@ -1,125 +1,140 @@
 .c-tooltip {
```
Comment (sebastianthulin) [inline source/sass/component/_tooltip.scss:1]: Define local css variables for colors. default to alpha color and contrasting color.

File: source/sass/component/_tooltip.scss [hunk 3]
```diff
@@ -1,125 +1,140 @@
...
+            animation-duration: .2s;
+        }
     }
 
-    &:before {
-        display: none;
-        content: "";
+    .c-tooltip__container {
         position: absolute;
+        border-radius: $border-radius-xs;
```
Comment (sebastianthulin) [inline source/sass/component/_tooltip.scss:22]: Create a local css var for this. Fallback to border-radiusxs.

File: source/sass/component/_tooltip.scss [hunk 4]
```diff
@@ -1,125 +1,140 @@
...
-                top: 50%;
-                right: -10px;
-                left: auto;
-                transform: translateY(-50%);
-
-                @include arrow("left", 5px, $color-black);
-            }
+    @include mq(map_get($breakpoints-map, "md")) {
+        .c-tooltip__container {
+            max-width: 400px;
```
Comment (sebastianthulin) [inline source/sass/component/_tooltip.scss:126]: Calculate from $base

File: source/sass/component/_tooltip.scss [hunk 5]
```diff
@@ -1,125 +1,140 @@
...
         }
     }
 
     @media print {
-
-        &:before {
+        .c-tooltip__container {
             display: none !important;
```
Comment (sebastianthulin) [inline source/sass/component/_tooltip.scss:132]: Is important required? This is not allowed outside utilities.

**Review by sebastianthulin — CHANGES_REQUESTED** [review]

File: source/sass/component/_tooltip.scss (hunk 5 above)
Comment (NiclasNorin) [reply source/sass/component/_tooltip.scss:132]: I do not believe it is. I kept this line from last implementation. I'll remove it!

File: source/js/tooltip.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                    tooltip.addEventListener(key, () => {
+                        this.handleHover(tooltip, container);
+                    })
+                });
+            });
+        }
+    }
+
+    handleLeave(tooltip, container) {
+        this.handleClasses(tooltip, ['c-tooltip--visible'], 'c-tooltip--hidden');
```
Comment (sebastianthulin) [inline source/js/tooltip.js:28]: MAy be changed to is-classes. Eg. c-tooltip.is-visible

File: source/sass/component/_tooltip.scss [hunk 7]
```diff
@@ -1,125 +1,132 @@
...
     }
 
-    &:before {
-        display: none;
-        content: "";
+    &.c-tooltip--visible {
+        .c-tooltip__container {
+            opacity:1;
+            display: block;
+            animation-name: opacityAnimation;
```
Comment (sebastianthulin) [inline source/sass/component/_tooltip.scss:19]: May be replaced with a transition instead.

**Review by sebastianthulin — CHANGES_REQUESTED** [review]

**Pull request merged**